account_banking_pain_base/models/account_payment_order.py

```diff
@@ -515,6 +522,17 @@ def generate_remittance_info_block(self, parent_node, line, gen_args):
                     line.communication_type
                 creditor_reference = etree.SubElement(
                     creditor_ref_information, 'CdtrRef')
+            # LSV+ ?
+            elif gen_args.get('pain_flavor') == 'pain.008.001.02.ch.03':
+                creditor_ref_info_type = etree.SubElement(
+                    creditor_ref_information, 'Tp')
+                creditor_ref_info_type_or = etree.SubElement(
+                    creditor_ref_info_type, 'CdOrPrtry')
+                creditor_ref_info_type_code = etree.SubElement(
+                    creditor_ref_info_type_or, 'Prtry')
+                creditor_ref_info_type_code.text = 'ESR'
+                creditor_reference = etree.SubElement(
+                    creditor_ref_information, 'Ref')
```

Comment:
Why not having this in module `l10n_ch` ? I don't understand the point to do it here as this should be the generic method that has no idea of the custom flavours that could be inherited. You can probably do this after calling super().